addons/payment_razorpay/tests/test_payment_provider.py

```diff
@@ -24,3 +25,27 @@ def test_signature_calculation_for_redirect_data(self):
         self.assertEqual(
             calculated_signature, '437b72e4e87362a39951b44487cf698410b074afdbed19ec44fffd32d2f863f3'
         )
+
+    def test_credentials_constraints(self):
+        """ Test that enabling a Razorpay provider without credentials raises a ValidationError,
+        and with credentials it succeeds.
+        """
+        provider_no_credentials = self.env['payment.provider'].create({
+            'name': 'Razorpay Missing Credentials',
+            'code': 'razorpay',
+            'state': 'test',
+        })
```

Comment:
Personal opinion, but I think it would be clearer (more obvious) if you'd clear the credentials of `self.provider` that is prepared in the test commons.